tests/aem_hw.pm

```diff
@@ -512,22 +502,30 @@ sub wait_for_startup {
 }
 
 sub handle_aem_startup {
-    # Workaround for the bug where video signal is missing
-    if (check_var("LUKSPASS_NO_VIDEO_WORKAROUND", "1")) {
-        sleep 15;
-        for my $i (1..10) {
-            if (check_screen("aem-good-secret", timeout => 10)){
-                last;
-            }
-            send_key 'esc';
-            send_key 'esc';
-            sleep 5;
-        }
-    }
+    lukspass_no_video_workaround("aem-good-secret");
     assert_screen "aem-good-secret", timeout => 180;
     send_key "ret";
 }
 
+# Workaround for the bug where video signal is missing
+sub lukspass_no_video_workaround {
+    my ($needle) = @_;
+
+    if (!check_var("LUKSPASS_NO_VIDEO_WORKAROUND", "1")) {
+        return;
+    }
+
+    sleep 15;
+    for my $i (1..10) {
+        if (check_screen($needle, timeout => 10)){
+            last;
+        }
+        send_key 'esc';
+        send_key 'esc';
```

Comment:
I planned to add a second `check_screen($needle, timeout => 1` between those two keys, it already happened to me twice that the platform ended up in text mode after both key presses. Could you add it, since you're reworking it anyway?

Reply:
Not quite what I had in mind, the second `esc` should be pressed conditionally, only if there is no match. I thought that the window is too small to cause any problems, but practice showed otherwise.